docker.go

```diff
@@ -804,26 +804,29 @@ func NewDockerProvider(provOpts ...DockerProviderOption) (*DockerProvider, error
 	}
 
 	// log docker server info only once
-	logOnce.Do(p.logDockerServerInfo)
+	logOnce.Do(func() {
+		logDockerServerInfo(context.Background(), p.client, p.Logger)
+	})
 
 	return p, nil
 }
 
-func (p *DockerProvider) logDockerServerInfo() {
+func logDockerServerInfo(ctx context.Context, client client.APIClient, logger Logging) {
 	infoMessage := `%v - Connected to docker: 
   Server Version: %v
   API Version: %v
   Operating System: %v
   Total Memory: %v MB
 `
 
-	info, err := p.client.Info(context.Background())
+	info, err := client.Info(ctx)
 	if err != nil {
-		p.Logger.Printf("failed getting information about docker server: %s", err)
+		logger.Printf("failed getting information about docker server: %s", err)
+		return
```

Comment:
I noticed this `if` was incomplete, as it should exit the function.

I wonder if we should raise an error here, as if the docker info failed, then we cannot continue. In the meantime, we can live with the return, as this function is called after the docker client creation, so it's expected using the client is healthy